apis/pipelines/v1alpha6/provider_types.go

```diff
@@ -33,10 +32,10 @@ type ProviderSpec struct {
 	Image         string `json:"image" yaml:"image"`
 	ExecutionMode string `json:"executionMode" yaml:"executionMode"`
 	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
-	ServiceAccount      string                           `json:"serviceAccount" yaml:"serviceAccount"`
-	DefaultBeamArgs     []apis.NamedValue                `json:"defaultBeamArgs,omitempty" yaml:"defaultBeamArgs,omitempty"`
-	PipelineRootStorage string                           `json:"pipelineRootStorage" yaml:"pipelineRootStorage"`
-	Parameters          map[string]*apiextensionsv1.JSON `json:"parameters,omitempty" yaml:"parameters,omitempty"`
+	ServiceAccount      string            `json:"serviceAccount" yaml:"serviceAccount"`
+	DefaultBeamArgs     []apis.NamedValue `json:"defaultBeamArgs,omitempty" yaml:"defaultBeamArgs,omitempty"`
+	PipelineRootStorage string            `json:"pipelineRootStorage" yaml:"pipelineRootStorage"`
+	Parameters          map[string]string `json:"parameters,omitempty" yaml:"parameters,omitempty"`
```

Comment:
We updated the type of `Provider.Spec.Parameters` [here](https://github.com/sky-uk/kfp-operator/pull/353/commits/cacab20dbe85f1f7ffdceccc1fc53ea9ea770f8b) from `map[string]string` to map[string]*JSON, but I'm not sure why.

- All parameters for the 2 providers we support are currently strings.
- Even if we did this to allow users to add params which are more complex structures without having to make code changes, that's not the case because we'd need to update the provider service code (where we redefine this `Parameters` struct)